src/stitching/delegateToSchema.ts

```diff
@@ -22,113 +22,148 @@ import FilterToSchema from '../transforms/FilterToSchema';
 import AddTypenameToAbstract from '../transforms/AddTypenameToAbstract';
 import CheckResultAndHandleErrors from '../transforms/CheckResultAndHandleErrors';
 
-export default async function delegateToSchema(
-  options: IDelegateToSchemaOptions,
-): Promise<any> {
-  const { info, args = {} } = options;
-  const rawDocument: DocumentNode = createDocument(
-    options.fieldName,
-    options.operation,
-    info.fieldNodes,
-    Object.keys(info.fragments).map(
-      fragmentName => info.fragments[fragmentName],
-    ),
-    info.operation.variableDefinitions,
+export function createBatchOperation(
+  targetSchema: GraphQLSchema,
+  targetOperation: 'query' | 'mutation' | 'subscription',
+  rootDefs: { [key: string]: [{ [key: string]: any }, { [key: string]: any }] },
+  graphqlContext: { [key: string]: any },
+  documentInfo: {
```

Comment:
Extract this to a type.

Reply:
Disregard that. Let's always pass info through rootDefs. Having two sources of truth is confusing.

Reply:
Also you don't have `fieldNodes` in `documentInfo` atm.

Reply:
It's kinda complicated. You might need to add fragment aliasing. But then we probably don't want to duplicate identical fragments. Same with variables.

Reply:
Yeah i always meant for that to be the full info object, but i just created a new type from the fields i was using. I'll update that 😄